src/sidebar/sidebarSlice.ts

```diff
@@ -105,19 +110,19 @@ const sidebarSlice = createSlice({
     addForm(state, action: PayloadAction<{ form: FormEntry }>) {
       const { form } = action.payload;
 
-      // Cancel pre-existing forms for the extension
-      void asyncForEach(state.forms, async (current) => {
-        if (current.extensionId === form.extensionId) {
-          const sender = await whoAmI();
-          await cancelForm({ tabId: sender.tab.id, frameId: 0 }, current.nonce);
-        }
-      });
-
-      state.forms = state.forms.filter(
-        (x) => x.extensionId !== form.extensionId
+      const [currentForms, notCurrentForms] = partition(
+        state.forms,
+        (x) => x.extensionId === form.extensionId
       );
-      // Unlike panels which are sorted, forms are like a "stack", will show the latest form available
-      state.forms.push(form);
+
+      // The UUID must be fetched synchronously to ensure the `form` Proxy element doesn't expire
```

Comment:
Adding a link to this PR

```suggestion
      // The UUID must be fetched synchronously to ensure the `form` Proxy element doesn't expire
      // https://github.com/pixiebrix/pixiebrix-extension/pull/4251
```